packages/ymax-planner/src/resolver.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  makeAgoricNames,
+  type SmartWalletKit,
+  type VstorageKit,
+} from '@agoric/client-utils';
+import type { SigningStargateClient } from '@cosmjs/stargate';
+import type { ExecuteOfferAction } from '@agoric/smart-wallet/src/smartWallet.js';
+import { submitAction } from './swingset-tx.ts';
+
+type SubmitOfferParams = {
+  walletKit: SmartWalletKit;
+  stargateClient: SigningStargateClient;
+  address: string;
+  vstorageKit: VstorageKit;
+  offerArgs?: object;
+  proposal?: object;
+};
+
+export const resolveSubscription = async ({
+  offerArgs = {},
+  proposal = {},
+  walletKit,
+  stargateClient,
+  address,
+  vstorageKit,
+}: SubmitOfferParams) => {
+  const { fromBoard, vstorage } = vstorageKit;
+  const { instance } = await makeAgoricNames(fromBoard, vstorage);
+
+  const action: ExecuteOfferAction = harden({
```

Comment:
doesn't this get replaced by the new non-offer-based invocation?  That's merged.